packages/core/src/helpers/warnDeprecatedProps.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect } from "react";
+
+type DeprecatedProps = Record<string, string>;
+
+export function useWarnDeprecatedProps<T extends Record<string, any>>(
```

Comment:
you made it generic but haven't used it as generic in any of the usages, is it needed?